### ISSUE TITLE
Sometimes pods FailedCreatePodSandBox on initial install of cluster

### ISSUE THREAD

**Issue by dustymabe** [openshift/okd#271]
**Describe the bug**

Sometimes when installing OKD I'll notice pods get stuck in `ContainerCreating`. If i describe the pods they have an error message about creating a sandbox failing:
```
Warning  FailedCreatePodSandBox  2m6s   kubelet, okd-node1  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_controller-manager-d2wtw_openshift-controller-manager_0fdcd086-6ca3-4922-95c1-4b6750eccdf6_0(7cd187081b8a2e462c8d07e89e50a40d87 │
│ e88550b11e598b43ebaaaf36e0787d): netplugin failed with no error message
```

WORKAROUND: If I reboot the node that is having trouble creating containers then the containers start to get created and the install progresses.

**Version**

VERSION: `4.5.0-0.okd-2020-07-14-153706-ga`
UPI: DigitalOcean

**How reproducible**
Not always, but it seems to happen often enough. I'm using hardware that is limited (8G instances) and pulling out some of the operators from the cluster (monitoring), etc. I'm mostly opening this issue to see if others are hitting the same problem. The workaround is to reboot the nodes one at a time and then the containers will start fine. 

**Log bundle**

https://dustymabe.fedorapeople.org/20200719-must-gather.tar.gz

**Comment by vrutkovs** [comment]
Seems to be some SDN issue:

>remote_runtime.go:105] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_installer-2-okd-node0_openshift-kube-apiserver_69072aa2-638c-4f1d-b820-53690d72477a_0(3088c44cc810fd5ee9f0f0ff3fadfd416390e9a9855cf2186503c9131735ab16): Multus: [openshift-kube-apiserver/installer-2-okd-node0]: error adding container to network "openshift-sdn": delegateAdd: error invoking confAdd - "openshift-sdn": error in getting result from AddNetwork: CNI request failed with status 400: 'the server was unable to return a response in the time allotted, but may still be processing the request (get pods installer-2-okd-node0)

**Comment by vrutkovs** [comment]
Tracked in bugzilla - https://bugzilla.redhat.com/show_bug.cgi?id=1851540

**Comment by dustymabe** [comment]
I also see this sometimes with OVN too:

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_apiserver-69c6857678-7dfst_openshift-apiserver_afdcd99a-c206-4d68-9c57-9bb3fca12ad9_0(d6013e6a6d89f86e03c549819a8c53b5403c6ee28bdf335d8d914923a42b346d): Multus: [openshift-apiserver/apiserver-69c6857678-7dfst]: error adding container to network "ovn-kubernetes": delegateAdd: error invoking confAdd - "ovn-k8s-cni-overlay": error in getting result from AddNetwork: CNI request failed with status 400: '[openshift-apiserver/apiserver-69c6857678-7dfst] failed to configure pod interface: timed out dumping br-int flow entries for sandbox: timed out waiting for the condition
```

**Comment by kai-uwe-rommel** [comment]
See https://github.com/openshift/okd/issues/274
Just happened once again.

**Comment by Mr-iX** [comment]
Had exactly the same problem with ovn kubernetes (version 4.5.0-0.okd-2020-08-12-020541).
A reboot fixed the issue after the installation. Seems like a race condition in the installation. Is onv kubernetes really production ready?

**Comment by kai-uwe-rommel** [comment]
A reboot (of the entire cluster) does not help in my case.
I have to trash and rebuild the entire cluster. :-(

**Comment by kai-uwe-rommel** [comment]
Will someone from development look into this?

**Comment by kai-uwe-rommel** [comment]
Example:

Events:                                                                                                                                                                                                                                                                             
  Type     Reason                  Age                  From                             Message                                                                                                                                                                                    
  ----     ------                  ----                 ----                             -------                                                                                                                                                                                    
  Normal   AddedInterface          15d                  multus                           Add eth0 [10.131.0.15/23]                                                                                                                                                                  
  Normal   Pulled                  15d                  kubelet, worker-03.devqs.ars.de  Container image "quay.io/openshift/okd-content@sha256:fe0e253849c6a771d10ec1f41c2fed0d95180cb77febd56f5d5ee21f639758dc" already present on machine                                         
  Normal   Created                 15d                  kubelet, worker-03.devqs.ars.de  Created container grafana                                                                                                                                                                  
  Normal   Started                 15d                  kubelet, worker-03.devqs.ars.de  Started container grafana                                                                                                                                                                  
  Normal   Pulled                  15d                  kubelet, worker-03.devqs.ars.de  Container image "quay.io/openshift/okd-content@sha256:644bcaca0a108801e83f4c3585c267e069bd6d7975c4234cd6498d96693211b0" already present on machine                                         
  Normal   Created                 15d                  kubelet, worker-03.devqs.ars.de  Created container grafana-proxy                                                                                                                                                            
  Normal   Started                 15d                  kubelet, worker-03.devqs.ars.de  Started container grafana-proxy                                                                                                                                                            
  Warning  FailedCreatePodSandBox  2m2s (x49 over 12m)  kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = error creating pod sandbox with name "k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0": layer not known

**Comment by jomeier** [comment]
Maybe you are also trapping in a problem with podman.

Try to ssh in the node that has the erronous pod running (or not running) and do a simple:

```
sudo podman images
```

if you see something like "layer not known" error messages, than podman/crio are broken.

You can delete /var/lib/containers/overlay-images and after a reboot it should work again. If that was the error.

**Comment by kai-uwe-rommel** [comment]
Yes, I immediately saw such "layer not known" error messages.
But there is no /var/lib/containers/overlay-images ... you mean /var/lib/containers/storage/overlay-images perhaps?
This exists. Deleted this. But now I see other errors:

  Warning  FailedCreatePodSandBox  8h (x71 over 8h)        kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = error creating pod sandbox with name "k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0": layer not known
  Warning  FailedCreatePodSandBox  8m53s (x487 over 114m)  kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = error creating pod sandbox with name "k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0": layer not known
  Warning  FailedCreatePodSandBox  87s                     kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = DeadlineExceeded desc = context deadline exceeded
  Warning  FailedCreatePodSandBox  75s                     kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = reserving pod sandbox name: error reserving pod name k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0 for id dfcad93eebbb4da1071b17ea5bbf9153a89f4e73762b037a7dd0a15ef5f72285: name is reserved
  Warning  FailedCreatePodSandBox  63s                     kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = reserving pod sandbox name: error reserving pod name k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0 for id a7a8ca47cff18efad8cd4dabf599b81761f8d634d16d30021be939c40692e6a4: name is reserved
  Warning  FailedCreatePodSandBox  50s                     kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = reserving pod sandbox name: error reserving pod name k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0 for id 33bce171ec665074f8bc0f67317dd3d4e8d941bc0bb4bb634c9fc77fc032b0fe: name is reserved
  Warning  FailedCreatePodSandBox  36s                     kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = reserving pod sandbox name: error reserving pod name k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0 for id 3ffb19c5c74e4ca7d7ef66924c5b392d570a67d70c8fe957a526cadca53cefc9: name is reserved
  Warning  FailedCreatePodSandBox  24s                     kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = reserving pod sandbox name: error reserving pod name k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0 for id 971522421d1d65dc3974001a031bf6e337b0798b304eb3562b43af2b0a7b81cc: name is reserved
  Warning  FailedCreatePodSandBox  9s                      kubelet, worker-03.devqs.ars.de  Failed to create pod sandbox: rpc error: code = Unknown desc = reserving pod sandbox name: error reserving pod name k8s_grafana-ff56fd59c-z7hdx_openshift-monitoring_23d28aab-1226-481f-aa3e-b58605370a66_0 for id 05094514f146525b73ca07d64e5c671dc198387e875febd60cdedbf3fc5e37a0: name is reserved

**Comment by kai-uwe-rommel** [comment]
Something is severely broken in the OKD / FCOS combination ... :-(

**Comment by jomeier** [comment]
After restarting some workers on the newest release of OKD 4 with ovn-kubernetes I have this error too.

**Comment by jomeier** [comment]
A side effect from this error is the one described here and should be fixed in in the next release of OKD 4.5 :

https://bugzilla.redhat.com/show_bug.cgi?id=1875282

**Comment by jomeier** [comment]
This link describes how to roll back from ovn-kubernetes to openshiftsdn network plugin:

https://docs.openshift.com/container-platform/4.5/networking/ovn_kubernetes_network_provider/rollback-to-openshift-sdn.html

I've done so and it worked. ovn-kubernetes seems still to be a little hot for me.

**Comment by kai-uwe-rommel** [comment]
OpenShiftSDN is also available with OKD?

**Comment by jomeier** [comment]
Sure. Since a few releases the standard plugin is ovn-kubernetes. That one also is necessary for windows containers available on 4.6/4.7.

openshiftsdn is mature. ovn-kubernetes seems to be tech preview.

**Comment by jomeier** [comment]
You can overwrite the standard network plugin in your install-config.yaml or afterwards with the instructions in my comment above.
